include/pm/device.h

```diff
@@ -113,6 +113,8 @@ struct pm_device {
 	enum pm_device_state state;
 	/** Device PM action callback */
 	pm_device_action_cb_t action_cb;
+	/** Power Domain it belongs */
+	const struct device *domain;
```

Comment:
nit: domains could be optional (to allow for resource optimization if not required)

Reply:
yep, that can be done